### DISCUSSION TITLE
Install Issue With Docker On TrueNAS Scale

### DISCUSSION THREAD

**Discussion by hodyhq** [amidaware/tacticalrmm#968] [Q&A]
**Server Info (please complete the following information):**
 - OS: TrueNAS Scale
 - Browser: Crave
 - RMM Version (as shown in top left of web UI): Not Installed Yet

**Installation Method:**
  - [ ] Standard
  - [X ] Docker

**Agent Info (please complete the following information):**
- Agent version (as shown in the 'Summary' tab of the agent from web UI):N/A
- Agent OS: [e.g. Win 10 v2004, Server 2012 R2]: N/A

**Describe the bug**
When trying to run docker-compose command: "sudo docker-compose up -d" I get the following error:
ERROR: The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.

Continue with the new image? [yN]y
Pulling tactical-nats (tactical-nats:latest)...

ERROR: pull access denied for tactical-nats, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

**To Reproduce**
1. Run sudo docker-compose up -d to install tacticalrmm

**Expected behavior**
TacticalRMM installs 

**Screenshots**
![Screenshot 2022-02-10 010003](https://user-images.githubusercontent.com/99109793/153347257-3e8be25b-82f8-4f97-84ff-54943a9144fe.png)

**Additional context**
Installed with Reverse Proxy. SSL certificate on proxy

**Comment by wh1te909** [comment]
looks like it's not picking up your `.env` file did you set that up? those variables need to be defined there
https://docs.tacticalrmm.com/install_docker/#5-setting-up-the-environment

**Reply by hodyhq** [reply]
So I decided to try the standard install. Will have a server to do it with next week.
But would you recommend Ubuntu desktop, Debian, or Ubuntu server?

**Reply by hodyhq** [reply]
And is there instructions for a proxy? Will be running TacticalRMM it on a VM.

**Reply by hodyhq** [reply]
And what specs would you recommend.

CPU count
Core count
Thread count
Ram

?

**Reply by silversword411** [reply]
Reading thru the docs will answer all those questions and more.

**Comment by wh1te909** [comment]
desktop OS is not supported. only ubuntu 20.04 LTS server or debian 10/11 server are supported
see https://docs.tacticalrmm.com/install_server/ for supported OS and minimum requirements
1 cpu core / 2gb ram is minimum
you don't need a proxy, tacticalrmm installs proxy for you.
if you mean you'll be running it on a VM on a hypervisor that's behind NAT and you want to expose it publicly then you just need to port forward in your router

**Comment by hodyhq** [comment]
I installed it on a VM - Ubuntu server 20.04.03
It launched and looks amazing.
But if I try to connect to an agent it never works.
"Unable to connect web socket, click to reconnect."
That's the error I get. 
I tried to do a recover connection, & recover mesh.
But it doesn't work.

Any ideas?

**Reply by wh1te909** [reply]
what's output of `sudo systemctl status meshcentral`

**Reply by hodyhq** [reply]
should I post everything it outputted?

**Reply by hodyhq** [reply]
● meshcentral.service - MeshCentral Server
     Loaded: loaded (/etc/systemd/system/meshcentral.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2022-02-14 04:32:19 UTC; 27min ago
   Main PID: 32068 (node)
      Tasks: 22 (limit: 4602)
     Memory: 121.1M
     CGroup: /system.slice/meshcentral.service
             ├─32068 /usr/bin/node node_modules/meshcentral
             └─32090 /usr/bin/node /meshcentral/node_modules/meshcentral --launch 32068

Feb 14 04:32:19 tacticalrmmserver systemd[1]: Started MeshCentral Server.
Feb 14 04:32:20 tacticalrmmserver node[32068]: MeshCentral HTTP redirection server running on port 1024.
Feb 14 04:32:20 tacticalrmmserver node[32068]: MeshCentral v0.9.67, WAN mode, Production mode.
Feb 14 04:32:20 tacticalrmmserver node[32068]: MeshCentral Intel(R) AMT server running on sub.domain.com:4433.
Feb 14 04:32:20 tacticalrmmserver node[32068]: MeshCentral HTTP server running on port 4430, alias port 443.
Feb 14 04:32:20 tacticalrmmserver node[32068]: Loaded web certificate from "https://sub.domain.com:443/", host: "sub.domain.com"
Feb 14 04:32:20 tacticalrmmserver node[32068]:   SHA384 cert hash: 235e4a8fe942f992c58721d8a9e8e5486659e0d4e9cd6d20a84cb5adc2e1620755de251bf32a36377e2470833dd8196b
Feb 14 04:32:20 tacticalrmmserver node[32068]:   SHA384 key hash: 45d978f3ddef825a465c7c17961cc951a62e4a2dbbe3eda586342a3eb28e0d8426c7bcd6a79ab46e560cd88b20d83274

**Reply by hodyhq** [reply]
that what it posted.
I just changed the domains for privacy

**Reply by hodyhq** [reply]
Do you think you have an idea why its doing that?

brief description of my environment:

Tacticalrmm running on ubuntu VM server on truenas scale.
port 4222 is directly forwarded
port 80 & 433 are forwarded through a Synology NAS domain proxy using the rmm, mesh, & API domains.

**Reply by wh1te909** [reply]
so meshcentral is working but issue is with the proxy. tacticalrmm already install nginx as a reverse proxy for you so there's no need to have another proxy in front that's probably the issue. just do a direct connection from your workstation to the truenas VM

**Reply by wh1te909** [reply]
if you need to use that synology proxy then you'll need to look through these docs https://docs.tacticalrmm.com/unsupported_scripts/ and change mesh config

**Reply by hodyhq** [reply]
It worked! Thank you so much!

Here are the details on what I did, in case you wanted or if anyone else wants.

I used the info from the link you provided:
https://docs.tacticalrmm.com/unsupported_scripts/#haproxy
And set the "CertUrl" to point to the IP address of my NAS that was running the reserve proxy and had the certificates installed.

Then I went into my Synology NAS  reverse proxy portal and added websockets to the rmm domains, especially the mesh domain under custom headers, by just clicking the create button and choosing WebSocket:

![Screenshot 2022-02-14 231305](https://user-images.githubusercontent.com/99109793/153991426-7410f5e6-42f4-4fbb-ac35-72ad22882512.png)

![Screenshot 2022-02-14 231348](https://user-images.githubusercontent.com/99109793/153991484-28e45b8d-d6fc-460e-a0bc-95733aed685a.png)

after that, it worked amazingly!!

Thank you so much!!

**Reply by silversword411** [reply]
Added to docs: https://github.com/amidaware/trmm-docs/commit/b8282a02cb45865dd5e71af703525c0e2c0a54bc

**Reply by hodyhq** [reply]
Awesome! Hopefully it helps anyone else encountering similar issues
